crates/secret-service-client/src/lib.rs

```diff
@@ -191,8 +191,12 @@ pub async fn make_v1_req(
     let archived = rkyv::access::<ArchivedVersionedServerMessage, rancor::Error>(&buf)
         .map_err(ClientError::DeserializationError)?;
 
-    let VersionedServerMessage::V1(msg) =
+    let VersionedServerMessage::V1(srv_msg) =
         deserialize(archived).map_err(ClientError::DeserializationError)?;
 
-    Ok(msg)
+    if let ServerMessage::TryAgain = srv_msg {
+        return Box::pin(make_v1_req(conn, msg, timeout_dur)).await;
+    }
```

Comment:
This can stack overflow on enough errors right? If so we may want to do a debug log before retrying again just so we have some sort of breadcrumb trail to follow if we blow the stack.